### DISCUSSION TITLE
Custom Models and Custom Dataset

### DISCUSSION THREAD

**Discussion by orlandito24** [mindee/doctr#1403] [Q&A]
I'm trying to create an ocr_predictor to work on my custom data, but I'm not sure where to start. I'm not sure if I should be using custom models, or if I'm able to adjust the current models, and tweak parameters. 

I saw this section, but I'm not sure what the weights would be or how the file would look?

`# Load custom detection and recognition model
det_model = db_resnet50(pretrained=False, pretrained_backbone=False)
det_model.load_weights("<path_to_checkpoint>/weights")
reco_model = crnn_vgg16_bn(pretrained=False, pretrained_backbone=False)
reco_model.load_weights("<path_to_checkpoint>/weights")
predictor = ocr_predictor(det_arch=det_model, reco_arch=reco_model, pretrained=False)`

**Comment by felixdittrich92** [comment]
Hi @orlandito24 : 👋,

If you have a dataset for chinese you can train your own recognition model see:

https://github.com/mindee/doctr/tree/main/references/recognition

**Reply by orlandito24** [reply]
Hi @felixdittrich92 I read this page and I don't understand what the difference would be between the train_set and val_set, would the images folder contain .jpgs of random chinese characters, or just grab cropped portions of the bigger chinese message? And would the json be what the .jpgs contain?

**Reply by felixT2K** [reply]
train_set is used to train the model and val_set to validate the model results while training.
the images folder contains in your case image word crops of chinese words and the labels.json the corresponding word which is present on the word crop as string

For example an image with the word "Hello" -> labels.json:

```
{
    "img_1.jpg": "Hello",
 }
 ```
 
 (One labels.json file for all images in the images folder) and each image has to contain 1 word not a text line or something else :)

**Reply by orlandito24** [reply]
Hi @felixT2K, I'm not sure how the train_set and the val_set would differ, since they seem to contain the same information? Would the val_set just be complete images of chinese words instead of cropped? Also would these custom models/data set be accessing the internet, or is it internal throught the repository?

**Reply by felixT2K** [reply]
No the structure of both sets is identical. But the content should differ.

The model will "see" the train set and the val set is "unseen" from the model to measure the models capacity
Mostly you can use a 80/20 -split

For example: 80000 train samples in the train set and 20000 val samples in the val set

**Reply by felixT2K** [reply]
Without having a dataset you can try to generate a synthetic one with: https://github.com/clovaai/synthtiger
(max 32 chinese symbols per image)
